### ISSUE TITLE
symbol lookup error: /usr/lib64/libGLX_mesa.so.0: undefined symbol: xcb_dri3_get_supported_modifiers

### ISSUE THREAD

**Issue by msjyoo** [knapsu/plex-media-player-appimage#17]
I'm getting this error on the latest update of openSUSE Tumbleweed.

Plex_Media_Player_2.18.0.893-48795f25_x64.AppImage

```
/tmp/.mount_Plex_MbEiu1Z/usr/bin/plexmediaplayer: symbol lookup error: /usr/lib64/libGLX_mesa.so.0: undefined symbol: xcb_dri3_get_supported_modifiers
```

```
objdump -T /usr/lib64/libGLX_mesa.so.0 | grep xcb_dri3_get_supp
0000000000000000      DF *UND*  0000000000000000              xcb_dri3_get_supported_modifiers
0000000000000000      DF *UND*  0000000000000000              xcb_dri3_get_supported_modifiers_window_modifiers
0000000000000000      DF *UND*  0000000000000000              xcb_dri3_get_supported_modifiers_reply
0000000000000000      DF *UND*  0000000000000000              xcb_dri3_get_supported_modifiers_screen_modifiers
```

**Comment by skandix** [comment]
i'm experiencing the same problem on debian testing (buster).

aswell as this error.
```
/tmp/.mount_Plex_MfHOx8H/usr/bin//plexmediaplayer.wrapper: line 125: 28791 Aborted                 LD_LIBRARY_PATH="" zenity --question --title="$TITLE" --text="$TEXT" 2> /dev/null
/tmp/.mount_Plex_MfHOx8H/usr/bin//plexmediaplayer.wrapper: line 125: 28794 Aborted                 LD_LIBRARY_PATH="" zenity --question --title="$TITLE" --text="$TEXT" 2> /dev/null
```

**Comment by probonopd** [comment]
This symbol was added only very recently (in 2018) to `libxcb-dri3.so.0.0.0`:

https://abi-laboratory.pro/index.php?view=compat_report&l=libxcb&v1=1.12&v2=1.13&obj=8520d&kind=abi#Added

What happens if you bundle none of the files mentioned on https://abi-laboratory.pro/index.php?view=objects_report&l=libxcb&v1=1.12&v2=1.13 inside the AppImage?

**Comment by knapsu** [comment]
Hi @msjyoo  and @skandix. Sorry for late response. Been busy with other projects.

Can you check if  version [2.20.0.909-46413dd1](https://knapsu.eu/data/plex/Plex_Media_Player_2.20.0.909-46413dd1_x64.AppImage) works for you?
It does not contain the mentioned `libxcb*.so` libraries.

**Comment by probonopd** [comment]
```
user@host ~ $ cat /etc/os-release 
NAME="Clear Linux OS"
VERSION=1
ID=clear-linux-os
ID_LIKE=clear-linux-os
VERSION_ID=26200
PRETTY_NAME="Clear Linux OS"
ANSI_COLOR="1;35"
HOME_URL="https://clearlinux.org"
SUPPORT_URL="https://clearlinux.org"
BUG_REPORT_URL="mailto:dev@lists.clearlinux.org"
PRIVACY_POLICY_URL="http://www.intel.com/privacy"
```

Fails to run with

```
user@host ~ $ '/home/user/Downloads/Plex_Media_Player_2.20.0.909-46413dd1_x64.AppImage'
/tmp/.mount_Plex_Mnaso4S/usr/bin/plexmediaplayer: symbol lookup error: /usr/lib64/libEGL.so.1: undefined symbol: gbm_bo_get_bpp
```

Can get it to launch by also removing `libgbm.so.1` from the bundle.

But then I get:

![certificates](https://user-images.githubusercontent.com/2480569/48318911-de5eaa80-e5fe-11e8-8752-17218f8455d6.png)

Clear Linux OS does not seem to have any certificates in those places:

```
user@host ~ $ strace -f squashfs-root/AppRun 2>&1 |  grep cert
[pid  1933] read(4</proc/1910/cmdline>, "grep\0cert\0", 2047) = 10
[pid  1912] access("/etc/ssl/certs/ca-certificates.crt", R_OK <unfinished ...>
[pid  1912] access("/etc/pki/tls/certs/ca-bundle.crt", R_OK) = -1 ENOENT (No such file or directory)
[pid  1912] access("/usr/share/ssl/certs/ca-bundle.crt", R_OK) = -1 ENOENT (No such file or directory)
[pid  1912] access("/usr/local/share/certs/ca-root-nss.crt", R_OK) = -1 ENOENT (No such file or directory)
[pid  1912] access("/etc/ssl/cert.pem", R_OK) = -1 ENOENT (No such file or directory)
```

Fwiw, this system does have lots of certificates in `/usr/share/ca-certs/.prebuilt-store/`.

**Comment by syco** [comment]
> Hi @msjyoo and @skandix. Sorry for late response. Been busy with other projects.
> 
> Can you check if version [2.20.0.909-46413dd1](https://knapsu.eu/data/plex/Plex_Media_Player_2.20.0.909-46413dd1_x64.AppImage) works for you?
> It does not contain the mentioned `libxcb*.so` libraries.

Hi,
I'm having the same issue on kubuntu 18.10, I can confirm that 2.20 is the last version that works for me.
Thanks

**Comment by knapsu** [comment]
@syco as a workaround please check if Flatpak pacakge works better for you.

Run the following command in a terminal:
`flatpak install https://knapsu.eu/data/plex/tv.plex.PlexMediaPlayer.flatpakref`

**Comment by probonopd** [comment]
Looks like removing libgbm.so.1 from the bundle is part of the solution, and using an SSL library that will look in the various places that distributions to happen to place certificates in, along [those](https://github.com/darealshinji/vlc-AppImage/issues/1#issuecomment-321041496) lines.

**Comment by syco** [comment]
Hi, here just to let you know that Plex_Media_Player_2.24.0.924-63fcaa8e_x64.AppImage works for me.
Thanks for the update

**Comment by syco** [comment]
Sorry to reopen, but Plex_Media_Player_2.25.0.940-485e2ea4_x64.AppImage reintroduced this issue

**Comment by syco** [comment]
flatpak works fine, i'll use that from now on.
Maybe you should put it top of the list in your website, as far as I can see there's no reason to prefer the appimage.
Thanks

**Comment by probonopd** [comment]
>  as far as I can see there's no reason to prefer the appimage

Providing an [AppImage](http://appimage.org/) has, among others, these advantages:
- Applications packaged as an AppImage can run on many distributions (including Ubuntu, Fedora, openSUSE, CentOS, elementaryOS, Linux Mint, and others) _without_ the need to install something like Flatpak or Snappy first
- One app = one file = super simple for users: just download one AppImage file, [make it executable](http://discourse.appimage.org/t/how-to-make-an-appimage-executable/80), and run
- No unpacking or installation necessary
- No root needed
- No system libraries changed
- Works out of the box, no installation of runtimes needed
- Optional desktop integration with `appimaged`
- Optional binary delta updates, e.g., for continuous builds (only download the binary diff) using AppImageUpdate
- Can optionally GPG2-sign your AppImages (inside the file)
- Works on Live ISOs
- Can use the same AppImages when dual-booting multiple distributions
- Can be listed in the [AppImageHub](https://appimage.github.io/) central directory of available AppImages
- Can double as a self-extracting compressed archive with the `--appimage-extract` parameter
- No repositories needed. Suitable/optimized for air-gapped (offline) machines

**Comment by JosephCW** [comment]
Wanted to chime in that I'm having the same problems using the .appimage on Ubuntu 18.10 running kernel 5.0.0. Flatpak doesn't seem to open anything when I try to run it. No error popup window either.

**Comment by mhogerheijde** [comment]
Same issue on Ubuntu 19.04, which moved to kernel 5.0.0.

```
$ uname -r
5.0.0-13-generic

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 19.04
Release:        19.04
Codename:       disco

$ ./Plex_Media_Player_2.27.0.949-542ba3ed_x64.AppImage
/tmp/mhogerheijde/.mount_Plex_M1xra2O/usr/bin/plexmediaplayer: symbol lookup error: /usr/lib/x86_64-linux-gnu/libGLX_mesa.so.0: undefined symbol: xcb_dri3_get_supported_modifiers

```

**Comment by probonopd** [comment]
Please see https://github.com/maxrd2/subtitlecomposer/issues/117

**Comment by probonopd** [comment]
Others seem to have those issues too, so it's not AppImage specific:
https://forums.autodesk.com/t5/eagle-forum/can-t-run-eagle-on-debian-10-testing/td-p/8312348

**Comment by MeanderingCode** [comment]
Using `Plex_Media_Player_2.32.0.973-62b2e27f_x64.AppImage`, I receive the same `symbol lookup error: /usr/lib/x86_64-linux-gnu/LibGLX_mesa.so.0: undefined symbol: xcb_dri3_get_supported_modifiers` error on Debian testing and I have the libraries suggested in the solution on the autodesk forum. Additionally, the permissions on `QtWebEngineProcess` are already `755` in the AppImage, as mentioned by the second solution.

The subtitlecomposer issue seems to suggest this was fixed there in December. I am guessing that is a dependency? These seem to suggest the issue should be resolved, but it does not seem to be.

Given that, I am not sure how to proceed with this. I would prefer to use the AppImage rather than the Flatpak.

**Comment by probonopd** [comment]
Did the solution given in https://github.com/maxrd2/subtitlecomposer/issues/117#issue-386437932 work?

**Comment by MeanderingCode** [comment]
Deleting `libxcb-dri3.so.0` and running from the extracted directory as in https://github.com/probonopd/linuxdeployqt/issues/331#issuecomment-441544377 worked.

Is there a reason the AppImage you are building and distributing is not using fixed version of libxcb-dri3 or excluding it?

**Comment by nbrownus** [comment]
Same deal as @MeanderingCode, removing those libs allows plex to start. Was hoping this would fix #21 but it does not :(

**Comment by knapsu** [comment]
Hi. Sorry for a long delay. I was offline for the last couple of weeks.

Committed the fix in 9f911a97da99438c1b558f24f4a4a033e38a01ec 

Please check if [this build](https://knapsu.eu/data/plex/Plex_Media_Player_20190515_36e9306d_x64.AppImage) works for you.

**Comment by MeanderingCode** [comment]
Works, thanks!
